### ISSUE TITLE
Accessibility tools can't access to headers info

### ISSUE THREAD

**Issue by rawlmp** [Sage/carbon#5386]
### Current behaviour

When user is tabbing page with NVDA tool column headers cannot be reached and read.

Note: Only `sortable` headers can be reached.

### Expected behaviour

Tabbing can highlight and NVDA can read all important elements of displayed page.

### CodeSandbox or Storybook URL

https://codesandbox.io/s/keen-hermann-7i90ji

### JIRA Ticket (Sage Only)

SBS-35310

### Suggested Solution

Possibility to reach all headers with NVDA tool for [keep a11y](https://www.w3.org/WAI/tutorials/tables/). 

### Carbon Version

109.3.0

### Design Tokens Version

2.40.0

### What browsers are you seeing the problem on?

Chrome, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS, Windows

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by edleeks87** [comment]
@sianford would you mind looking into this before we raise an issue?

**Comment by rawlmp** [comment]
> @sianford would you mind looking into this before we raise an issue?

Could you contact @negreanugc as he has all the tools needed?

**Comment by negreanugc** [comment]
Hi @sianford, Hi @edleeks87 ! I have reproduced this issue using the voice over tool(MacOS). Also, tested this against Carbon web page and we have the same behaviour.

**Comment by Parsium** [comment]
FE-5326

**Comment by nineteen88** [comment]
This doesn't appear to be an issue. I believe there is some confusion between `Browse` mode and `Focus` mode of screen readers. You can navigate the table as expected in `Browse` mode - it will read out the titles and the content properly.

For more information on the difference between the two modes this is a good explainer:  https://www.accessibility-developer-guide.com/knowledge/screen-readers/desktop/browse-focus-modes/
